_rules/SC4-1-2-role-has-required-states-and-properties.md

```diff
@@ -0,0 +1,131 @@
+
+However, there is a bit uncertainty about the extent of support for implicit values for roles, since the same requirement is only a SHOULD in the the [Core Accessibility API Mappings 1.1](https://www.w3.org/TR/core-aam-1.1/#statePropertyMappingGeneralRules): 
+> "Where the author has not provided values for required attributes, user agents SHOULD process as if the default value was provided."
+
+WAI-ARIA also prescribes on [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) that:
+> “Content authors MUST provide a non-empty value for required states and properties. Content authors MUST NOT use the value undefined for required states and properties, unless undefined is an explicitly-supported value of that state or property." 
+
+The conflicting information around user agent support for implicit values for roles has been raised with the ARIA Working Group as https://github.com/w3c/aria/issues/787. Until the Auto-WCAG Community Group gets clarification from the ARIA Working Group, we assume for this rule, that relying on the default values is enough to satisfy this accessibility requirement.
+
+See more about the possible implications of any lacking user agent support for implicit values for roles in the ”Accessibility Support” section for this rule.
+
+## Accessibility Support
+
+If browsers and/or assistive technologies do not support leaving out [required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState) if an [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole) is specified in [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole), passing this rule may not be enough to avoid accessibility issues due to missing required states and properties. Testing with assistive technologies have however showed that there seems to be general support for implicit values for roles.
+
+**Note:** The required states and properties that could be impacted by this can be found in the Core Accessibility API Mappings 1.1 [Overview of default values for missing required attributes](https://www.w3.org/TR/core-aam-1.1/#authorErrorDefaultValuesTable).
+
+## Background
+
+- [ARIA5: Using WAI-ARIA state and property attributes to expose the state of a user interface component](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/ARIA5)
+- [WAI-ARIA required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState)
+-  [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Element has required states (no properties required for this role)
+
+```html
+<div role="option" aria-selected="false"></div>
+```
+
+#### Passed example 2
+
+Implicit value for roles: ´option´ role has an implicit value (´false´) specified for the required state ´aria-selected´
```

Comment:
Lets have a case where the attribute is defined without a value, and one with only whitespace.

Reply:
Test cases added